adding tracking for Gutenberg and non-Gutenberg posts on open

mzorz opened the pull request:
This PR adds tracking to see how many opens on Gutenberg posts are there on the WordPress Android app.

Note there is redundant code shamelessly copied from [the Editor library project](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java#L555-L576), as it's easier to keep this way for tracking purposes only.

mzorz commented:
cc @diegoreymendez as it may be good to have the same tracks naming? feel free to propose other strings if you feel like so

koke commented:
I think I'd rather have the Gutenberg detection as a property rather than two separate new events, and we can reuse existing events.

I'm looking at this on iOS because I know where to look:

- Open existing post: `post_list_button_pressed` with property `button=edit`, there seems to be no similar event in Android
- Save a post: there are several different ones: `editor_draft_saved`, `editor_post_scheduled`, `editor_post_published`, `editor_post_updated`. iOS also has `editor_quick_post_published`, `editor_quick_draft_saved`.
- Creating a post: `editor_post_created`. But if you just create an empty new post, it's not going to be a Gutenberg one 😉 

We can just add a `has_gutenberg_blocks` property to all of those events

koke commented:
Opened a similar issue for iOS in https://github.com/wordpress-mobile/WordPress-iOS/issues/9580

mzorz commented:
@koke addressed as suggested 👍 
- added new `post_list_button_pressed` event on list item press event 17442a3
- added tracking to `editor_draft_saved, editor_post_scheduled, editor_post_published, editor_post_updated` on cd52d9d, 1f46dfe, d0ecf6b and c8facaa (and 839043c)

Ready for review

daniloercoli commented on WordPress/src/main/java/org/wordpress/android/ui/posts/PostPreviewActivity.java:
```diff
@@ -284,7 +287,9 @@ private void revertPost() {
             if (!mPost.isLocalDraft()) {
-                AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.EDITOR_UPDATED_POST, mSite);
+                Map<String, Object> properties = new HashMap<>();
+                properties.put("has_gutenberg_blocks", PostUtils.contentContainsGutenbergBlocks(mPost.getContent()));
```
Niptick: what do you think on use a string constant for  `has_gutenberg_blocks` property?

mzorz replied:
Addressed in 9f08330d66

mzorz commented:
> contentContainsGutenbergBlocks is used both in Aztec and here. What about moving this routine to wp-utils library?

As per chat, we'll leave this as an improvement separate from this PR for now 👍 

Ready for another round @daniloercoli 🙇

daniloercoli commented:
Looks good! Feel free to merge once Travis is happy.

diegoreymendez commented:
> contentContainsGutenbergBlocks is used both in Aztec and here. What about moving this routine to wp-utils library?

Or even better have it in the API.  It sounds weird to me that each platform is duplicating this logic and maybe we can tackle this as a mobile mini-project?

Just a thought.

hypest commented:
> Or even better have it in the API

👋 @diegoreymendez , we'd still need it locally too, to use it while offline, right?

diegoreymendez commented:
@hypest - I meant in the post data, so that if you have the post you also have the flag at all times.
